pandas/io/stata.py

```diff
@@ -2205,7 +2220,13 @@ def write_file(self):
             self._write_file_close_tag()
             self._write_map()
         finally:
-            self._file.close()
+            try:
```

Comment:
can you move to a close method (or _close if needed), or is this the only place where things are needing to be closed? can you add some comments